src/libcore/ptr.rs

```diff
@@ -653,6 +685,38 @@ impl<T: ?Sized> *mut T {
             Some(&mut *self)
         }
     }
+
+    /// Calculates the distance between two pointers. The returned value is in
+    /// units of T: the distance in bytes is divided by `mem::size_of::<T>()`.
+    ///
+    /// If the address different between the two pointers ia not a multiple of
+    /// `mem::size_of::<T>()` then the result of the division is rounded towards
+    /// zero.
+    ///
+    /// If `T` is a zero-sized type then it is instead treated as having a size
+    /// of 1. That is, the returned value is simply the arithmetic difference
+    /// of the two pointer values.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// let a = [0; 5];
+    /// let ptr1: *mut i32 = &mut a[1];
+    /// let ptr2: *mut i32 = &mut a[3];
+    /// assert_eq!(ptr1.offset_to(ptr2), 2);
+    /// assert_eq!(ptr2.offset_to(ptr1), -2);
+    /// assert_eq!(ptr1.offset(2), ptr2);
+    /// assert_eq!(ptr2.offset(-2), ptr1);
+    /// ```
+    #[unstable(feature = "offset_to", issue = "0")]
+    #[inline]
+    pub fn offset_to(self, other: *const T) -> isize where T: Sized {
```

Comment:
Well it doesn't really matter whether the pointer is mutable or not, and `*const T` accepts both.